core-overlay.html

```diff
@@ -111,666 +124,373 @@
-       * @type Object
-       * @default the target element
-       */
-      sizingTarget: null,
-    
-      /**
-       * Set opened to true to show an overlay and to false to hide it.
-       * A `core-overlay` may be made initially opened by setting its
-       * `opened` attribute.
-       * @attribute opened
-       * @type boolean
-       * @default false
-       */
-      opened: false,
+        /**
+         * If true, the ESC key does not cancel the overlay.
+         *
+         * @attribute noCancelOnEscKey
+         * @type boolean
+         * @default false
+         */
+        noCancelOnEscKey: {
+          type: Boolean,
+          value: false
+        },
 
-      /**
-       * If true, the overlay has a backdrop darkening the rest of the screen.
-       * The backdrop element is attached to the document body and may be styled
-       * with the class `core-overlay-backdrop`. When opened the `core-opened`
-       * class is applied.
-       *
-       * @attribute backdrop
-       * @type boolean
-       * @default false
-       */    
-      backdrop: false,
+        /**
+         * If true, clicking outside the overlay does not cancel the overlay.
+         *
+         * @attribute noCancelOnOutsideClick
+         * @type boolean
+         * @default false
+         */
+        noCancelOnOutsideClick: {
+          type: Boolean,
+          value: false
+        }
 
-      /**
-       * If true, the overlay is guaranteed to display above page content.
-       *
-       * @attribute layered
-       * @type boolean
-       * @default false
-      */
-      layered: false,
-    
-      /**
-       * By default an overlay will close automatically if the user
-       * taps outside it or presses the escape key. Disable this
-       * behavior by setting the `autoCloseDisabled` property to true.
-       * @attribute autoCloseDisabled
-       * @type boolean
-       * @default false
-       */
-      autoCloseDisabled: false,
+      },
+
+      get backdropElement() {
+        return this._backdrop;
+      },
+
+      registered: function() {
+        this._backdrop = document.createElement('div');
+        this._backdrop.className = 'core-overlay-backdrop';
+      },
+
+      ready: function() {
+        this.addElementListener(this, 'click', 'onClick');
+        this.ensureTargetSetup();
+      },
+
+      detached: function() {
+        this.opened = false;
+        this.completeBackdrop();
+        manager.removeOverlay(this);
+      },
 
       /**
-       * By default an overlay will focus its target or an element inside
-       * it with the `autoFocus` attribute. Disable this
-       * behavior by setting the `autoFocusDisabled` property to true.
-       * @attribute autoFocusDisabled
-       * @type boolean
-       * @default false
+       * Toggle the opened state of the overlay.
+       *
+       * @method toggle
        */
-      autoFocusDisabled: false,
+      toggle: function() {
+        this.opened = !this.opened;
```

Comment:
I tend to locate mutation to a single place if possible. In this case if the overlay is opened you want to close it and vice versa:

```js
toggle: function() {
  if (this.opened) { this.close(); }
  else { this.open(); }
}
```